File: backend/app/schemas/employee.py
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime
from typing import Optional

class EmployeeBase(BaseModel):
    employee_id: str = Field(..., description="Unique Employee ID")
    full_name: str
    email: EmailStr
    department: str

class EmployeeCreate(EmployeeBase):
    pass

class Employee(EmployeeBase):
    id: str
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/employee.ts
export interface Employee {
    id: string; // Internal UUID
    employee_id: string; // Human-readable ID
    full_name: string;
    email: string;
    department: string;
    created_at?: string;
}

export interface EmployeeCreate {
    employee_id: string;
    full_name: string;
    email: string;
    department: string;
}
